src/index.js

```diff
@@ -1,28 +1,117 @@
 import './index.css';
 
-const tasks = [
-  { description: 'Learn Webpack', completed: true, index: 0 },
-  { description: 'Finish quiz', completed: false, index: 1 },
-  { description: 'Learn supplication', completed: false, index: 2 },
-  { description: 'Wash utensils', completed: true, index: 3 },
-];
+let tasks = [];
+if (localStorage.getItem('tasks')) {
+  tasks = JSON.parse(localStorage.getItem('tasks'));
+}
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+function addTask(description) {
+  const task = {
+    description,
+    completed: false,
+    index: tasks.length,
+  };
+  tasks.push(task);
+  saveTasks();
+}
+function updateIndexes() {
+  for (let i = 0; i < tasks.length; i += 1) {
+    tasks[i].index = i;
+  }
+}
+
+function deleteTask(index) {
+  tasks.splice(index, 1);
+  updateIndexes();
+  saveTasks();
+}
 
 function displayTasks() {
   const todoList = document.getElementById('todo-list');
   todoList.innerHTML = '';
   tasks.sort((a, b) => a.index - b.index);
 
-  tasks.forEach((task) => {
+  tasks.forEach((task, index) => {
     const listItem = document.createElement('li');
     listItem.innerHTML = `
-      <input type="checkbox" ${task.completed ? 'checked' : ''} />
-      <span class="taskdesc">${task.description}</span>
-      <i class="bi bi-three-dots-vertical"></i>
-    `;
+          <input type="checkbox" ${task.completed ? 'checked' : ''} />
+          <span class="taskdesc ${task.completed ? 'strikethrough' : ''}">${task.description}</span>
+          <!--<span class="taskdesc">${task.description}</span>-->
+          <i class="bi bi-pencil-square edit-task"></i>
+          <i class="bi bi-trash delete-task"></i>
+        `;
     todoList.appendChild(listItem);
+
+    // Add event listener to checkbox input
+    const checkbox = listItem.querySelector('input[type="checkbox"]');
+    checkbox.addEventListener('change', () => {
+      task.completed = checkbox.checked;
+      saveTasks();
+      displayTasks();
+    });
+
+    // Add event listener to edit icon
+    const editIcon = listItem.querySelector('.edit-task');
+    editIcon.addEventListener('click', () => {
+      const taskDesc = listItem.querySelector('.taskdesc');
+      const input = document.createElement('input');
+      input.type = 'text';
+      input.value = taskDesc.innerText;
+      input.classList.add('edit-input');
+      taskDesc.replaceWith(input);
+
+      // Add event listener to edit input to save changes
+      input.addEventListener('keydown', (e) => {
+        if (e.key === 'Enter') {
+          const newDesc = input.value.trim();
+          if (newDesc.length > 0) {
+            task.description = newDesc;
+            taskDesc.innerText = newDesc;
+            input.replaceWith(taskDesc);
+            saveTasks();
+          }
+        }
```

Comment:
- [ OPTIONAL] Good work implementing the edit feature. I suggest you make the input field for the edit action consistent with the display tasks
![edit](https://user-images.githubusercontent.com/99656687/236137987-5932ab8c-b7c4-42cc-9fb2-0c55aabb989c.png)